src/modules/oAuth/oAuth.controller.ts

```diff
@@ -0,0 +1,22 @@
+import { GoogleOAuthGuard } from './guards/google-oauth.guard';
+import { Controller, Get, Request, Response, UseGuards } from '@nestjs/common';
+import { OAuthService } from './oAuth.service';
+
+@Controller('auth')
+export class OAppController {
+  constructor(private readonly appService: OAuthService) {}
+
+  @Get('google/login')
+  @UseGuards(GoogleOAuthGuard)
+  async googleAuth() {}
+
+  @Get('google/callback')
+  @UseGuards(GoogleOAuthGuard)
+  async googleAuthRedirect(@Request() req, @Response() res) {
```

Comment:
You can also create User Decorator that will extract user object from request and also check if it is present.